packages/catalog-process/src/services/readModelService.ts

```diff
@@ -146,27 +151,73 @@ export function readModelServiceBuilder(
         ],
       };
 
+      const descriptorsFilter =
+        states.length === 0 && attributesIds.length === 0
```

Comment:
may I suggest to split the filter for states and attributes like in Scala code?
This way we reduce the coupling between parameters and the query complexity.
(probably the third level of `elemMatch` is not required)

Reply:
Updated